### PR TITLE
Add rudimentary Pillow stubs

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by Akuli** [comment]
```
>>> from PIL import ImageTk
>>> ImageTk
<module 'PIL.ImageTk' from '/tmp/e/lib/python3.8/site-packages/PIL/ImageTk.py'>
```

But the stub doesn't contain `ImageTk.py`. Are other submodules missing too? Is this intentional?

**Comment by srittau** [comment]
Yes, it's very incomplete. (But it has the required `__getattr__()` in `__init__()`).

**Comment by Akuli** [comment]
So in a stub, having `__getattr__` in `foobar/__init__.py` also affects code like `from foobar.submodule import thingy`, even though that's not how it works at runtime?

**Comment by srittau** [comment]
I thought it does, but I could be wrong. I will add empty stubs if that's not the case.

**Comment by srittau** [comment]
I just checked with mypy and the following passes with these stubs:

```python
from PIL.X import Y

reveal_type(Y)  # Revealed type is "Any"
```

File: stubs/Pillow/PIL/ImageDraw.pyi [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    ) -> tuple[int, int, int, int]: ...
+
+def Draw(im: Image, mode: str | None = ...) -> ImageDraw: ...
+
+Outline: Any
+
+@overload
+def getdraw(im: None = ..., hints: Container[Literal["nicest"]] | None = ...) -> tuple[None, Any]: ...
+@overload
+def getdraw(im: Image, hints: Container[Literal["nicest"]] | None = ...) -> tuple[Image, Any]: ...
```
Comment (Akuli) [inline stubs/Pillow/PIL/ImageDraw.pyi:147]: Are you sure `tuple[Foo, Bar]` is not affected by the mypy bug?

Comment (srittau) [reply stubs/Pillow/PIL/ImageDraw.pyi:147]: It's only `tuple[X, ...]` and `tuple` (as well as others) in type aliases.

**Comment by JelleZijlstra** [comment]
Fixed the merge conflict by removing it from the pyright exclude list.

**Review by Akuli — APPROVED** [review]

**Comment by jakebailey** [comment]
Wish I had seen this; I have stubbed out quite a bit of PIL on our stub staging repo: https://github.com/microsoft/python-type-stubs/tree/main/PIL

**Pull request merged**